diff --git a/src/archive/archive.cpp b/src/archive/archive.cpp
index 43aefac5..b04e9544 100644
--- a/src/archive/archive.cpp
+++ b/src/archive/archive.cpp
@@ -446,7 +446,7 @@ int main(int argc, const char **argv) {
             stat(path, &s);
 
             // Windows uses mtime which is a time_t rather than a struct with nanoseconds
-            #ifdef _WIN32
+            #if defined(_WIN32) || defined(__APPLE__)
             archive_entry_set_mtime(entry, s.st_mtime, 0);
             #else
             archive_entry_set_mtime(entry, s.st_mtim.tv_sec, 0);
diff --git a/src/tag/parser/compile/model.cpp b/src/tag/parser/compile/model.cpp
index 487a82b9..7143fb39 100644
--- a/src/tag/parser/compile/model.cpp
+++ b/src/tag/parser/compile/model.cpp
@@ -3,6 +3,7 @@
 #include <invader/build/build_workload.hpp>
 #include <invader/tag/parser/parser.hpp>
 #include <invader/tag/parser/compile/model.hpp>
+#include <climits>
 
 namespace Invader::Parser {
     template<typename M> static void postprocess_hek_data_model(M &what) {
